src/libstore/gc.cc

```diff
@@ -781,9 +781,19 @@ void LocalStore::collectGarbage(const GCOptions & options, GCResults & results)
         for (auto & path : topoSortPaths(visited)) {
             if (!dead.insert(path).second) continue;
             if (shouldDelete) {
-                invalidatePathChecked(path);
-                deleteFromStore(path.to_string());
-                referrersCache.erase(path);
+                try {
+                    invalidatePathChecked(path);
+                    deleteFromStore(path.to_string());
+                    referrersCache.erase(path);
+                } catch (Error & e) {
```

Comment:
I don't think we should ignore arbitrary errors, since they'll probably cascade or repeat (e.g. SQLite errors or disk full conditions). It's probably better to identity specific error conditions and handle them appropriately.